File: atividade-11/atividade.go
package main

import (
	"fmt"
)

type AFD struct {
	state string
}

func NewAFD() *AFD {
	return &AFD{state: "q0"} 
}

func (afd *AFD) transition(c rune) {
	if afd.state == "q0" {
		if c == 'a' {
			afd.state = "q1" 
		}
	} else if afd.state == "q1" {
		if c == 'a' {
			afd.state = "q0" 
		}
	}
}

func (afd *AFD) isAccepted(input string) bool {
	afd.state = "q0" 
	for _, c := range input {
		afd.transition(c)
	}
	return afd.state == "q1" 
}

func main() {
	afd := NewAFD()
	fmt.Println(afd.isAccepted("a"))   
	fmt.Println(afd.isAccepted("aa"))  
	fmt.Println(afd.isAccepted("ab"))  
	fmt.Println(afd.isAccepted("bba")) 
	fmt.Println(afd.isAccepted("bb"))  
}

File: atividade-11/atividade.py
class AFD:
    def __init__(self):
        self.state = 'q0'  

    def transition(self, char):
        if self.state == 'q0':
            if char == 'a':
                self.state = 'q1'  
            elif char == 'b':
                self.state = 'q0'  
        elif self.state == 'q1':
            if char == 'a':
                self.state = 'q0'  
            elif char == 'b':
                self.state = 'q1'  

    def is_accepted(self, string):
        self.state = 'q0'  
        for char in string:
            self.transition(char)
        return self.state == 'q1'  

if __name__ == '__main__':
    afd = AFD()
    print(afd.is_accepted("a"))       
    print(afd.is_accepted("aa"))      
    print(afd.is_accepted("ab"))     
    print(afd.is_accepted("bba"))     
    print(afd.is_accepted("bb"))      
    print(afd.is_accepted("aaa"))      